types/react-native/index.d.ts

```diff
@@ -6037,9 +6037,7 @@ interface PlatformWebStatic extends PlatformStatic {
     OS: 'web';
 }
 
-declare const OpaqueColorValue: unique symbol;
-type OpaqueColorValue = typeof OpaqueColorValue;
-
+type OpaqueColorValue = symbol & { __TYPE__: 'Color' };
```

Comment:
This needs to still have `unique`, right? Otherwise someone could pass in any symbol, which would be invalid, right?

Reply:
The `{ __TYPE__: 'Color'}` part of the `ColorValue` type will make it so that you can't just pass any symbol. If you try something like

```
const Test = () => (
    <View
        style={{
            backgroundColor: Symbol('test'),
        }}
    />
);
```

that won't work. This is one of the suggested ways to approximate the opaque typing in Flow in TS (ex. [here](https://github.com/niieani/typescript-vs-flowtype#opaque-type-alias)).

However this would be allowed:

```
type TestType = symbol & {readonly __TYPE__: 'Color'};
const Test = () => (
    <View
        style={{
            backgroundColor: Symbol('test') as TestType,
        }}
    />
);
```

If this is an issue, then I would need to go back to the drawing board and find a different solution, and I'd be happy to hear suggestions for other things to try. The issue I'm having with using unique symbol is that you can't assign it from an object, and so if you have objects that hold color information as a bunch of properties that map to ColorValue (like theming objects), you end up needing to cast a lot of objects to force things to work.

For what it's worth, this works with the current definition, so I'm not sure if the above being allowed is much worse:

```
const Test = () => (
    <View
        style={{
            backgroundColor: Symbol('test') as ColorValue,
        }}
    />
);
````

Reply:
Sounds legit to me as a user!

Reply:
Any additional thoughts in response to my comments?

Reply:
I don't know TypeScript well enough to be able to really review this. I agree that this seems to fix a thing that *should* be able to be used as a color. My biggest concern would be if this allows additional things as colors that shouldn't be. I'm not sure what those would be though. 

I'll approve this though, worst case we can fix forward if needed.

Reply:
Thanks, I appreciate it! It's a fair concern, if a fix is necessary I can help out with fixing it.